tests/tests/tier0/bluechi-change-port-with-c-cmd-option/test_bluechi_change_port_with_c_cmd_option.py

```diff
@@ -0,0 +1,67 @@
+# SPDX-License-Identifier: LGPL-2.1-or-later
+
+from typing import Dict
+import os
+
+from bluechi_test.test import BluechiTest
+from bluechi_test.container import BluechiControllerContainer, BluechiNodeContainer
+from bluechi_test.config import BluechiControllerConfig, BluechiNodeConfig
+from bluechi_test.util import read_file
+
+NODE_FOO = "node-foo"
+
+
+def check_if_node_and_ctrl_connected(ctrl, node):
+    # ctrl_log = ctrl.exec_run("systemctl status bluechi-controller.service")[1]
+    # node_log = node.exec_run("systemctl status bluechi-agent.service")[1]
+    # if "port=8421" in node_log and "Connected to manager as" in node_log \
+    #    and "Registered managed node" in ctrl_log:
+    #     return True
+    # return False
+    result, output = ctrl.exec_run(f'bluechictl status {NODE_FOO} bluechi-agent.service')
+    print("!!!!!!!!!!!!!!!!!!!!!!!!!!!!")
+    print(result)
+    print(str(output).split('\n')[2].split('|')[2].strip() == 'active')
+    assert result == 0
+    assert str(output).split('\n')[2].split('|')[2].strip() == 'active'
+
+
+def exec(ctrl: BluechiControllerContainer, nodes: Dict[str, BluechiNodeContainer]):
+    node_foo = nodes[NODE_FOO]
+    config_file_location = "/var/tmp"
+    bluechi_agent_str = "bluechi-agent"
+    bluechi_controller_str = "bluechi-controller"
+    file_location_ctrl = os.path.join("config-files", "ctrl_port_8421.conf")
+    file_location_agent = os.path.join("config-files", "agent_port_8421.conf")
+
+    # Copying relevant config files into the nodes container
+    content = read_file(file_location_agent)
+    node_foo.create_file(config_file_location, file_location_agent, content)
+    content = read_file(file_location_ctrl)
+    ctrl.create_file(config_file_location, file_location_ctrl, content)
+
+    ctrl.restart_with_config_file(
+        os.path.join(config_file_location, "ctrl_port_8421.conf"), bluechi_controller_str)
```

Comment:
Personally after a restart I'd add a check if bluechi-controller is really listening on a different port using netstat or lsof within the controller container